### PR TITLE
patch coreDNS config instead of replacing it outright

### PR DESCRIPTION
**Pull request by BenTheElder**
this should handle updates to the default config and multiple versions better

### CODE REVIEW

**Comment by BenTheElder** [comment]
cc @aojea

**Comment by neolit123** [comment]
LGTM on my end, `sed` is definitely less readable but this is better in the long term.

@aojea had a PR for this today:
https://github.com/kubernetes-sigs/kind/pull/799

so i'm leaving the lgtm to him.

**Comment by aojea** [comment]
I'm from my mobile, is it missing the ready plugin?

Definitely this approach looks better, we should move in this direction

**Comment by BenTheElder** [comment]
> I'm from my mobile, is it missing the ready plugin?

it will have whatever the default is, if this is not in the default we could patch it in.

**Comment by BenTheElder** [comment]
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/sigs.k8s.io_kind/803/pull-kind-conformance-parallel-ipv6/1164627475699863552/
```
I0822 20:11:48.466] Original CoreDNS config:
I0822 20:11:48.466] apiVersion: v1
I0822 20:11:48.466] data:
I0822 20:11:48.467]   Corefile: |
I0822 20:11:48.467]     .:53 {
I0822 20:11:48.467]         errors
I0822 20:11:48.467]         health
I0822 20:11:48.467]         ready
I0822 20:11:48.467]         kubernetes cluster.local in-addr.arpa ip6.arpa {
I0822 20:11:48.467]            pods insecure
I0822 20:11:48.467]            fallthrough in-addr.arpa ip6.arpa
I0822 20:11:48.467]            ttl 30
I0822 20:11:48.468]         }
I0822 20:11:48.468]         prometheus :9153
I0822 20:11:48.468]         forward . /etc/resolv.conf
I0822 20:11:48.468]         cache 30
I0822 20:11:48.468]         loop
I0822 20:11:48.468]         reload
I0822 20:11:48.468]         loadbalance
I0822 20:11:48.468]     }
I0822 20:11:48.468] kind: ConfigMap
I0822 20:11:48.469] metadata:
I0822 20:11:48.469]   creationTimestamp: "2019-08-22T20:11:12Z"
I0822 20:11:48.469]   name: coredns
I0822 20:11:48.469]   namespace: kube-system
I0822 20:11:48.469]   resourceVersion: "202"
I0822 20:11:48.469]   selfLink: /api/v1/namespaces/kube-system/configmaps/coredns
I0822 20:11:48.469]   uid: a7d7f788-df70-4d05-9a3a-f3eace6e1a87
I0822 20:11:48.490] Patched CoreDNS config:
I0822 20:11:48.490] apiVersion: v1
I0822 20:11:48.490] data:
I0822 20:11:48.491]   Corefile: |
I0822 20:11:48.491]     .:53 {
I0822 20:11:48.491]         errors
I0822 20:11:48.491]         health
I0822 20:11:48.491]         ready
I0822 20:11:48.491]         kubernetes cluster.local in-addr.arpa ip6.arpa {
I0822 20:11:48.491]            pods insecure
I0822 20:11:48.491]            ttl 30
I0822 20:11:48.491]         }
I0822 20:11:48.492]         prometheus :9153
I0822 20:11:48.492]         cache 30
I0822 20:11:48.492]         reload
I0822 20:11:48.492]         loadbalance
I0822 20:11:48.492]     }
I0822 20:11:48.492] kind: ConfigMap
I0822 20:11:48.492] metadata:
I0822 20:11:48.492]   creationTimestamp: "2019-08-22T20:11:12Z"
I0822 20:11:48.492]   name: coredns
I0822 20:11:48.493]   namespace: kube-system
I0822 20:11:48.493]   resourceVersion: "202"
I0822 20:11:48.493]   selfLink: /api/v1/namespaces/kube-system/configmaps/coredns
I0822 20:11:48.493]   uid: a7d7f788-df70-4d05-9a3a-f3eace6e1a87
```

**Comment by BenTheElder** [comment]
it has `ready`.

File: hack/ci/e2e.sh [hunk 1]
```diff
@@ -102,48 +102,36 @@ run_tests() {
...
+    # Get the current config
+    local original_coredns
+    original_coredns=$(kubectl get -oyaml -n=kube-system configmap/coredns)
+    echo "Original CoreDNS config:"
+    echo "${original_coredns}"
+    # Patch it
+    local fixed_coredns
+    fixed_coredns=$(
+      sed \
+        -e 's/^(\s)+.*cluster\.local.*$/\0kubernetes cluster.local internal in-addr.arpa ip6.arpa/' \
```
Comment (aojea) [inline hack/ci/e2e.sh]: @BenTheElder this line is not working, the output is not being modified :(

https://github.com/kubernetes-sigs/kind/pull/803#issuecomment-524063894

https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/81750/pull-kubernetes-conformance-kind-ipv6/1165220112794390528/build-log.txt

**Pull request merged**